### PR TITLE
The only invalid XML character we need to handle is the null character. XMLParser can handle all the others.

### PR DESCRIPTION
**Pull request by leonardr**
This branch addresses https://jira.nypl.org/browse/SIMPLY-1684. It takes a much simpler approach than we were using previously to parsing XML documents that contain XML characters -- take care of the null character ourselves and let `lxml.XMLParser` handle everything else.

### CODE REVIEW

File: util/http.py [hunk 1]
```diff
@@ -234,8 +234,8 @@ def _request_with_timeout(cls, url, make_request_with, *args, **kwargs):
 
         try:
             if verbose:
-                logging.info("Sending %s request to %s: kwargs %r",
-                             http_method, url, kwargs)
+                logging.info("Sending request to %s: args %r kwargs %r",
+                             url, args, kwargs)
```
Comment (leonardr) [inline util/http.py]: This is an unrelated problem I found while debugging. There's no variable `http_method` in this method -- it's one of the `args`.

File: util/xmlparser.py [hunk 2]
```diff
@@ -78,44 +48,20 @@ def int_of_optional_subtag(self, tag, name, namespaces=None):
             return v
         return int(v)
 
-    def scrub_xml(self, xml):
```
Comment (EdwinGuzman) [inline util/xmlparser.py]: "Scrub" is the xml term for cleaning the data, right?

Comment (leonardr) [reply util/xmlparser.py]: It's not official but yeah, it refers to taking some data that might contain invalid characters, and removing those characters.
